### PR TITLE
Implement BooksController#Show and BooksController#Index

### PR DESCRIPTION
**Pull request by cquiro**
## Summary

* Add and configure gems versionist, active_model_serializers and wor-paginate
* Create BooksController and BookSerializer
* Create endpoint for books#show
* Create endpoint for books#index (including pagination)
* Require authentication for all actions except the ones that are performed by the devise_controller

## Trello Card

https://trello.com/c/K1gUlUqv/23-crear-rutas-index-y-show-para-book

### CODE REVIEW

File: app/serializers/api/v1/book_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module Api
```
Comment (facucompe) [inline app/serializers/api/v1/book_serializer.rb]: No need to use `api/v1` folders

File: app/controllers/api/v1/books_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      include Wor::Paginate
```
Comment (facucompe) [inline app/controllers/api/v1/books_controller.rb]: Move this to application controller

File: app/controllers/api/v1/books_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      end
+
+      def show
+        render json: book, serializer: Api::V1::BookSerializer, status: :ok
+      end
+
+      private
+
+      def book
+        @book ||= Book.find(params[:id])
```
Comment (facucompe) [inline app/controllers/api/v1/books_controller.rb]: No need to use instance variable or memoization.

Comment (facucompe) [reply app/controllers/api/v1/books_controller.rb]: Also use a separate method to permit params.
```
def book_params
  params.permit(:id)
end
```

File: app/controllers/api/v1/books_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      end
+
+      def books
+        @books ||= Book.all
```
Comment (facucompe) [inline app/controllers/api/v1/books_controller.rb]: Same here. No need to use an instance variable and memoization

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@ class ApplicationController < ActionController::Base
 
   include DeviseTokenAuth::Concerns::SetUserByToken
 
+  before_action :authenticate_user!, unless: :devise_controller?
```
Comment (facucompe) [inline app/controllers/application_controller.rb]: We usually use an `api_controller` that inherits from `application_controller` where we put the authentication.

**Review by facucompe — CHANGES_REQUESTED** [review]

**Comment by facucompe** [comment]
Try using **spinal-case** for branch names instead of **snake_case**

**Review by lmasello — APPROVED** [review]
Just what facu asked and then it's good to go 👍

File: app/controllers/api/v1/books_controller.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+module Api
+  module V1
+    class BooksController < ApiController
+      def index
+        render_paginated books, each_serializer: Api::V1::BookSerializer, status: :ok
```
Comment (facucompe) [inline app/controllers/api/v1/books_controller.rb]: Watch out, you are not serializing. It should just be `BookSerializer`

**Review by facucompe — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/books_controller.rb (hunk 6 above)
Comment (cquiro) [reply app/controllers/api/v1/books_controller.rb]: @facucompe Done!

**Review by facucompe — APPROVED** [review]

**Pull request merged**